src/core/search/search.service.ts

```diff
@@ -0,0 +1,403 @@
+        },
+      });
+
+      const totalCount = body.hits.total.value;
+      const hits = body.hits.hits;
+      let data = hits.map((item: any) => item._source);
+      data = await this.formatedData(data);
+
+      return {
+        pagination: {
+          itemCount: body.length,
+          totalItems: totalCount,
+          itemsPerPage: limit,
+          totalPages: Math.ceil(totalCount / limit),
+          currentPage: page,
+          randomSeed: random,
+        },
+        profiles: data,
+      };
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  async parseAndPrepareData() {
+    const profiles = await this.getProfiles();
+
+    const body = profiles.flatMap((doc) => [
+      { index: { _index: 'profiles', _id: doc.id } },
+      {
+        id: doc.id,
+        userId: doc.userId,
+        name: doc.user.name,
+        lastname: doc.user.lastname,
+        email: doc.user.email,
+        description: doc.description,
+        mainTitle: doc.mainTitle,
+        countryResidence: doc.countryResidence,
+        skills: doc.skills,
+        experience: doc.experience,
+      },
+    ]);
+
+    return body;
+  }
+
+  async deleteIndex() {
+    const resp = await this.searchClient.indices.delete({
+      index: 'profiles',
+    });
+
+    return resp;
+  }
+
+  private async validateQueryArrayRelation(
+    relations: string[] | null,
+    relationName: string,
+  ) {
+    if (relations) {
+      relations = Array.isArray(relations) ? relations : [relations];
+
+      const validationQuery = await this.profileRepository.find({
+        relations: [relationName],
+        where: {
+          skills: {
+            name: In(relations),
+          },
+        },
+      });
+
+      if (!validationQuery) {
+        throw new BadRequestException(`Valor invaldo para ${relationName}`);
+      }
+    } else {
+      relations = null;
+    }
+
+    return relations;
+  }
+
+  private async validateQueryArrayCountry(column: string[] | null) {
+    if (column) {
+      column = Array.isArray(column) ? column : [column];
+
+      const validationQuery = await this.profileRepository.find({
+        where: {
+          countryResidence: In(column),
+        },
+      });
+
+      if (!validationQuery) {
+        throw new BadRequestException(`Valor invaldo para ${column}`);
+      }
+    } else {
+      column = null;
+    }
+
+    return column;
+  }
+
+  private async validateCarrera(carrera: any) {
+    if (carrera) {
+      carrera = Array.isArray(carrera) ? carrera : [carrera];
+
+      carrera = carrera.map((c: string) => {
+        const carreraLower = c.toLowerCase() as Career;
+
+        if (!Object.values(Career).includes(carreraLower)) {
+          throw new BadRequestException(
+            `Valor inválido para carrera: ${carreraLower}`,
+          );
+        }
+
+        return carreraLower;
+      });
+    } else {
+      carrera = null;
+    }
+
+    return carrera;
+  }
+
+  private async formatedData(data: any) {
```

Comment:
Por que chuchas el ASYNC XD???
```suggestion
  private formatedData(data: any) {
```